easypost/shipment.py

```diff
@@ -62,24 +66,46 @@ def label(self, **params) -> "Shipment":
 
     def lowest_rate(self, carriers: List[str] = None, services: List[str] = None) -> Rate:
         """Get the lowest rate of a shipment."""
-        carriers = carriers or []
-        services = services or []
-        lowest_rate = None
+        lowest_rate = Util._get_lowest_object_rate(self, carriers, services)
+
+        return lowest_rate
 
-        carriers = [c.lower() for c in carriers]
-        services = [service.lower() for service in services]
+    def lowest_smartrate(self, delivery_days: Optional[int] = None, delivery_accuracy: Optional[str] = None) -> Rate:
+        """Get the lowest smartrate of a shipment."""
+        smartrates = self.get_smartrates()
+        lowest_smartrate = self.get_lowest_smartrate(smartrates, delivery_days, delivery_accuracy)
```

Comment:
`get_lowest_smartrate` can be a static function (no need for `self.`) since it doesn't use anything from this particular instance of Shipment

Reply:
`get_lowest_smartrate` is already static? What do you mean here?

Reply:
There is a `self` in the first parameter.

Reply:
Not in `get_lowest_smartrate`. You guys are probably referencing `lowest_smartrate` which definitely needs self because you call it on the Shipment object.

Reply:
You call it here as `self.get_lowest_smartrate`

Reply:
Would you like me to call it with `Shipment.get_lowest_smartrate`? We need `self` for the line directly above it because we need to `get_smartrates` of the shipment object before we can get the lowest smartrate, so `self` will already be in use in this function and is required earlier.